core/server/common/src/main/java/alluxio/master/journal/AdvanceFuture.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.journal;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Used to track and cancel journal advancing.
+ */
+public class AdvanceFuture {
```

Comment:
Naming suggestion, does `catchup` make more sense here?

Reply:
For this class only or does you suggestion includes `JournalSystem.advance()` as well?

Reply:
Replace the concept of advance with catchup since we already have a concept of catchup.